internal/common/ttlmap/ttl_map.go

```diff
@@ -12,26 +12,36 @@ import (
 type TTLMap struct {
 	md            *ttlMapData
 	sweepInterval int64
+	done          chan struct{}
 }
 
 // New creates a TTLMap. The sweepIntervalSeconds arg indicates how often
 // entries are checked for expiration. The maxAgeSeconds arg indicates how long
 // entries can persist before getting evicted. Call Start() on the returned
 // TTLMap to begin periodic sweeps which check for expiration and evict entries
 // as needed.
-func New(sweepIntervalSeconds int64, maxAgeSeconds int64) *TTLMap {
+// done is the channel that will be used to signal to the timer to stop its work.
+func New(sweepIntervalSeconds int64, maxAgeSeconds int64, done chan struct{}) *TTLMap {
```

Comment:
Apologies for the size of this PR now. I originally wanted to simply create the channel inside of `New`, but an existing test `exporter/signalfxexporter/TestConfigGetMetricTranslator` does an `Assert.Equal` on objects that contain the TTLMap as a private member. The assertion fails as the pointers are not equal. I've attempted a few workarounds (`cmp` and `cmpopts`, yaml or JSON marshalling). Nothing works.

If the preference is to simply not have a done channel and use the timer and the `timer.Stop()` method instead, I can modify this PR to pass that in. Either way, the signature of `New` will need to be updated to accomodate.

Reply:
An alternative that may be less impacting would be to pass the done channel to Start. I don't think passing in the Ticker would work as it gets started at creation. I don't think it would be a great design to pass a running Ticker to these methods.

Reply:
the TTLMap struct might also create its own done channel. Wouldn't that be best?

Reply:
That would be best, but the usage and testing is where the challenge comes in. The `TestConfigGetMetricTranslator` is testing to ensure the config is created properly relative to the metric translator itself in the signalfx exporter. When the TTLMap creates its own channel the equality comparison fails.

The other options are remove the test, or add a public equality checking method to the metrics translator.